### PR TITLE
Eba_tips: used a function to generate tip content in order to avoid actually building it if not necessary.

### PR DESCRIPTION
**Pull request by sagotch**

### CODE REVIEW

**Comment by vasilisp** [comment]
Looks good. Have you updated the BeSport code to use this? We can merge it whenever you are ready.

**Comment by sagotch** [comment]
I do have a branch synchronized with this :+1:

File: src/eba_tips.eliom [hunk 1]
```diff
@@ -117,19 +117,20 @@ let%shared block ?(a = []) ~name ~content () =
...
     then Lwt.return None
     else begin
-      let close_button = Ot_icons.D.close () in
+      let box_ref = ref None in
+      let close = [%client (fun () ->
+        let () = match !(~%box_ref) with
+          | Some x -> Manip.removeSelf x
+          | None -> () in
+        Lwt.async (fun () -> set_tip_seen ~%name);
+        Lwt.return () : _ -> _) ] in
```
Comment (vasilisp) [inline src/eba_tips.eliom]: Upon further inspection: why does `close` need to return an Lwt value?

File: src/eba_tips.eliomi [hunk 2]
```diff
@@ -51,7 +51,8 @@ val bubble :
   ?width:int ->
   ?parent_node:[< `Body | Html_types.body_content ] Eliom_content.Html.elt ->
   name:string ->
-  content: Html_types.div_content Eliom_content.Html.elt list ->
+  content:((unit -> unit Lwt.t) Eliom_client_value.t
```
Comment (vasilisp) [inline src/eba_tips.eliomi]: Can you document the `(unit -> unit Lwt.t) Eliom_client_value.t` argument?

File: src/eba_tips.eliom (hunk 1 above)
Comment (sagotch) [inline src/eba_tips.eliom]: Very good question, I have no idea why the Lwt thing is here. Probably needed in a first version, I'll remove it!

File: src/eba_tips.eliomi (hunk 2 above)
Comment (sagotch) [inline src/eba_tips.eliomi]: Sure

File: src/eba_tips.eliom (hunk 1 above)
Comment (sagotch) [inline src/eba_tips.eliom]: Or maybe the question is: why does `close` is wrapped in a `Lwt.async` call? Shouldn't we wait for the action to complete?

i.e. replace 
```
Lwt.async (fun () -> set_tip_seen ~%name) ; Lwt.return () 
```
with
```
set_tip_seen ~%name
```

Comment (sagotch) [inline src/eba_tips.eliom]: I pushed a commit unwrapping the closing function call, is it okay to do that?

Comment (vasilisp) [inline src/eba_tips.eliom]: Thanks, looks good now: `close` truly does do something asynchronous, as opposed to just scheduling another asynchronous task.

**Pull request merged**